File: src_v3/danbooru/konachan_client.py
import random

from danbooru.danbooru_client import DanbooruClient


class KonachanClient(DanbooruClient):
    def _get_index_params(self):
        page = random.randint(1, 2431)
        return {"limit": "100", "page": str(page)}

    _title = "Konachan"
    _index_url = "https://konachan.com/post.json"

    def _get_full_post_url(self, post):
        return "https://konachan.com/post/show/" + str(post["id"])

    def _get_tag_list(self, post):
        return post["tags"].split()

File: src/main/kotlin/com/github/hannesbraun/katarina/modules/danbooru/KonachanClient.kt
package com.github.hannesbraun.katarina.modules.danbooru

import com.google.gson.JsonObject
import kotlinx.coroutines.CoroutineScope
import kotlin.random.Random

class KonachanClient(scope: CoroutineScope) : DanbooruClient(scope) {
    override val name = "Konachan"
    override val indexUrl = "https://konachan.com/post.json"

    override fun getIndexParameters(): List<Pair<String, String>> {
        val page = Random.nextInt(1, 2550)
        return listOf(Pair("limit", postBufferSize.toString()), Pair("page", page.toString()))
    }

    override fun generatePost(
        id: String, fileUrl: String, rating: String, score: String, tagList: List<String>
    ) = KonachanPost(id, fileUrl, rating, score, tagList)

    override fun getTagListFromJson(post: JsonObject): List<String> = post.getString("tags").split(" ")
}

class KonachanPost(id: String, fileUrl: String, rating: String, score: String, tagList: List<String>) : DanbooruPost(
    id,
    fileUrl,
    rating,
    score,
    tagList
) {
    override fun getFullPostUrl() = "https://konachan.com/post/show/$id"
}